sdk/ai/azure-ai-inference/tests/test_model_inference_client.py

```diff
@@ -267,6 +275,183 @@ def test_embeddings(self, **kwargs):
+        )
+        image_embedding_input = ModelClientTestBase._get_image_embeddings_input()
+        for _ in range(2):
+            try:
+                response = client.embed(
+                    input=[image_embedding_input], raw_request_hook=self.request_callback
+                )
+            except ServiceRequestError as _:
+                # The test should throw this exception!
+                self._validate_image_embeddings_json_request_payload()
+                continue
+            assert False
+    
+    # Regression test. Send a request that includes all supported types of input objects, with embeddings settings
+    # specified in the constructor and all of them overwritten in the 'embed' call.
+    # Make sure the resulting JSON payload that goes up to the service is the correct one after hand-inspection.
+    def test_image_embeddings_request_payload_with_defaults_and_overrides(self, **kwargs):
+        client = sdk.ImageEmbeddingsClient(
+            endpoint="http://does.not.exist",
+            credential=AzureKeyCredential("key-value"),
+            headers={"some_header": "some_header_value"},
+            user_agent="MyAppId",
+            dimensions=1024,
+            encoding_format=sdk.models.EmbeddingEncodingFormat.UINT8,
+            input_type=sdk.models.EmbeddingInputType.DOCUMENT,
+            model_extras={
+                "hey1": 2,
+                "key2": False,
+                "key3": "Some other value",
+                "key9": "Yet another value",
+            },
+            model="some-other-model-id",
+        )
+        image_embedding_input = ModelClientTestBase._get_image_embeddings_input()
+        for _ in range(2):
+            try:
+                response = client.embed(
+                    input=[image_embedding_input],
+                    dimensions=2048,
+                    encoding_format=sdk.models.EmbeddingEncodingFormat.UBINARY,
+                    input_type=sdk.models.EmbeddingInputType.QUERY,
+                    model_extras={
+                        "key1": 1,
+                        "key2": True,
+                        "key3": "Some value",
+                        "key4": [1, 2, 3],
+                        "key5": {"key6": 2, "key7": False, "key8": "Some other value", "key9": [4, 5, 6, 7]},
+                    },
+                    model="some-model-id",
+                    raw_request_hook=self.request_callback,
+                )
+            except ServiceRequestError as _:
+                # The test should throw this exception!
+                self._validate_image_embeddings_json_request_payload()
+                continue
+            assert False
+    
+    # **********************************************************************************
+    #
+    #                      HAPPY PATH SERVICE TESTS - IMAGE EMBEDDINGS
+    #
+    # **********************************************************************************
+    """
```

Comment:
I'm working on these new service tests. Will commit another change soon to enable them.

Reply:
Tests added. Recordings updated.